momepy/preprocessing.py

```diff
@@ -740,3 +746,327 @@ def _get_extrapolated_line(coords, tolerance, point=False):
     if point:
         return b
     return pygeos.linestrings([a, b])
+
+
+def _polygonize_ifnone(edges, polys):
+    if polys is None:
+        pre_polys = polygonize(edges.geometry)
+        polys = gpd.GeoDataFrame(geometry=[g for g in pre_polys], crs=edges.crs)
+    return polys
+
+
+def _selecting_rabs_from_poly(
+    gdf, circom_threshold=0.7, area_threshold=0.85, include_adjacent=True
+):
+    """
+    From a GeoDataFrame of polygons, returns a GDF of polygons that are
+    above the Circular Compaactness threshold.
+
+    Return
+    ________
+    GeoDataFrames : round abouts and adjacent polygons
+    """
+    # calculate parameters
+    gdf.loc[:, "area"] = gdf.geometry.area
+    circom_serie = CircularCompactness(gdf, "area").series
+
+    # selecting round about polygons based on compactness
+    rab = gdf[circom_serie > circom_threshold]
+    # exclude those above the area threshold
+    area_threshold_val = gdf.area.quantile(area_threshold)
+    rab = rab[rab["area"] < area_threshold_val]
+
+    if include_adjacent is True:
+        # calculating some parameters
+        bounds = rab.geometry.bounds
+        rab = pd.concat([rab, bounds], axis=1)
+        rab["deltax"] = rab.maxx - rab.minx
+        rab["deltay"] = rab.maxy - rab.miny
+        rab["rab_diameter"] = rab[["deltax", "deltay"]].max(axis=1)
+
+        # selecting the adjacent areas that are of smaller than itself
+        if GPD_10:
+            rab_adj = gpd.sjoin(gdf, rab, predicate="intersects")
+        else:
+            rab_adj = gpd.sjoin(gdf, rab, op="intersects")
+        rab_adj = rab_adj[rab_adj.area_right >= rab_adj.area_left]
+        rab_adj.index.name = "index"
+        rab_adj["hdist"] = 0
+
+        # adding a hausdorff_distance threshold
+        # TODO: (should be a way to verctorize)
+        for i, group in rab_adj.groupby("index_right"):
+            for g in group.itertuples():
+                hdist = g.geometry.hausdorff_distance(rab.loc[i].geometry)
+                rab_adj.loc[g.Index, "hdist"] = hdist
+
+        rab_plus = rab_adj[rab_adj.hdist < rab_adj.rab_diameter]
+
+    else:
+        rab["index_right"] = rab.index
+        rab_plus = rab
+
+    # only keeping relevant fields
+    geom_col = rab_plus.geometry.name
+    rab_plus = rab_plus[[geom_col, "index_right"]]
+
+    return rab_plus
+
+
+def _rabs_center_points(gdf, center_type="centroid"):
+    """
+    From a selection of roundabouts, returns an aggregated GeoDataFrame
+    per round about with extra column with center_type.
+    """
+    # creating a multipolygon per RAB (as opposed to dissolving) of the entire
+    # composition of the RAB
+    # temporary DataFrame where geometry is the array of pygeos geometries
+    # Hack until shapely 2.0 is out.
```

Comment:
```suggestion
    # Hack until shapely 2.0 is out.
    # TODO: replace pygeos with shapely 2.0
```

it is good to keep `# TODO` in there as I sometimes look for these and fix them.